execute/report/report.go

```diff
@@ -196,6 +197,29 @@ func (b *execReportBuilder) checkMessage(
 		"data", messageTokenData.ToByteSlice())
 
 	// 3. Check if the message has a valid nonce.
+	status := b.checkMessageNonce(msg, execReport)
+	if status != "" {
+		return execReport, status, nil
+	}
+
+	// 4. Check if the message is too costly to execute.
+	if slices.Contains(execReport.CostlyMessages, msg.Header.MessageID) {
```

Comment:
Consider: if `execReport.CostlyMessages` is a map, this could be a constant time lookup. I'm not sure how big these slices could be, I imagine it could be bad if there's a gas spike and there are 1000s of ccip messages across different chains.

Reply:
Good point, I'll use a map

Reply:
These are the costly messages per report, they won't be more than the message limit per commit report. I did try making CostlyMessages a map/set, but I ran into JSON encoding issues. We can revisit this if it becomes an issue.